### PR TITLE
 Add Hyper-threading aware scheduling test cases for PerformanceProfile.

### PR DESCRIPTION
**Pull request by dgonyier**
Test case IDs to execute based on current cluster toplogy:
 * SNO vs non-SNO
 * Hyper-Threading enabled vs disbled
 * SNO with Workload Partitioning enabled vs disabled

### CODE REVIEW

**Comment by dgonyier** [comment]
Looks like all of the tests failed due to this error:

```error: error creating buildah builder: initializing source docker://registry.build02.ci.openshift.org/ci/managed-clonerefs@sha256:9fc8b45f8a9f01e83a8fbbb8ea1f196b23375e971ebeb79e66d5861f2ea9ac0c: pinging container registry registry.build02.ci.openshift.org: Get "https://registry.build02.ci.openshift.org/v2/": dial tcp: i/o timeout```

**Comment by fromanirh** [comment]
/retest
logs suggest some flakiness

File: test/e2e/performanceprofile/functests/1_performance/cpu_management.go [hunk 1]
```diff
@@ -617,7 +789,9 @@ func promotePodToGuaranteed(pod *corev1.Pod) *corev1.Pod {
 
 func getTestPodWithAnnotations(annotations map[string]string, cpus int) *corev1.Pod {
 	testpod := pods.GetTestPod()
-	testpod.Annotations = annotations
+	if len(annotations) > 0 {
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/1_performance/cpu_management.go:792]: why do we need this?

File: test/e2e/performanceprofile/functests/1_performance/cpu_management.go [hunk 2]
```diff
@@ -563,9 +564,180 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
...
+		AfterEach(func() {
+			if testpod == nil {
+				return
+			}
+			deleteTestPod(testpod)
+		})
+
+		table.DescribeTable("Verify Hyper-Thread aware scheduling for guaranteed pods",
+			func(htDisabled bool, snoCluster bool, snoWP bool) {
+				// FIt("[test_id:46959] verify guaranteed pod uses CPU thread sibling pairs", func() {
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/1_performance/cpu_management.go:589]: please remove commented code (here and below)

File: test/e2e/performanceprofile/functests/1_performance/cpu_management.go [hunk 3]
```diff
@@ -563,9 +564,180 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
...
+	testpod.Namespace = testutils.NamespaceTesting
+
+	By(fmt.Sprintf("Creating test pod with %d cpus", cpuCount))
+	err := testclient.Client.Create(context.TODO(), testpod)
+	Expect(err).ToNot(HaveOccurred())
+	err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+	logEventsForPod(testpod)
+	Expect(err).ToNot(HaveOccurred(), "Start pod failed")
+	// Pod should not have status.reason SMTAlignmentError
+	// Pod should not start, but in case it does start for some reason...
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/1_performance/cpu_management.go:733]: please elaborate why the pod should NOT be starting

**Review by fromanirh — COMMENTED** [review]
initial review, I need to review carefully the `checkPodHTSiblings` function

File: test/e2e/performanceprofile/functests/1_performance/cpu_management.go (hunk 3 above)
Comment (dgonyier) [reply test/e2e/performanceprofile/functests/1_performance/cpu_management.go:733]: @fromanirh arbitrary check. A "should never happen" case. I'll remove that logic.

**Comment by fromanirh** [comment]
/approve
/lgtm
non-blocking comments, let's move forward

**Pull request merged**